lms/djangoapps/dashboard/sysadmin.py

```diff
@@ -585,7 +586,7 @@ def get(self, request, *args, **kwargs):
 
         course_id = kwargs.get('course_id')
         if course_id:
-            course_id = CourseKey.from_string(course_id)
+            course_id = CourseLocator.from_string(course_id)
```

Comment:
Also, it's very unusual to call `CourseLocator.from_string()` directly rather than letting `CourseKey.from_string()` delegate the call. Please add a comment explaining why we're doing it in this case.